### PR TITLE
Use confHook to cons to autogenModules

### PR DESCRIPTION
**Pull request by phadej**
ping @RyanGlScott

### CODE REVIEW

File: src/Distribution/Extra/Doctest.hs [hunk 1]
```diff
@@ -435,3 +446,33 @@ testDeps :: ComponentLocalBuildInfo -> ComponentLocalBuildInfo
...
+        testSuite = condTreeData condTree
+        bi = testBuildInfo testSuite
+        om = otherModules bi
+        am = autogenModules bi
+
+        -- append
+        om' = fromString "Build_doctests" : om
+        am' = fromString "Build_doctests" : am
+
+        bi' = bi { otherModules = om', autogenModules = am' }
```
Comment (RyanGlScott) [inline src/Distribution/Extra/Doctest.hs]: I'm a bit surprised that you have to add `Build_doctests` to both `other-modules` and `autogen-modules`! Is this by design?

Comment (phadej) [reply src/Distribution/Extra/Doctest.hs]: Unfortunaly yes. There is a ticket (IIRC) to add `autogen-modules` to `other-modules` automatically, if they aren't `exposed-modules`. But it's won't be there even for `cabal-version: 2.2`.

**Comment by RyanGlScott** [comment]
Looks good to me! I was able to successfully compile every package in the Kmettiverse with this version of `cabal-doctest`, and I successfully ran the doctests of a few of these packages as well, so I think this is ready to go.

**Comment by phadej** [comment]
@RyanGlScott this is on Hackage: https://hackage.haskell.org/package/cabal-doctest-1.0.5, if you want to add `-Werror`, go ahead! :)

**Pull request merged**